src/wp-admin/includes/class-wp-media-list-table.php

```diff
@@ -748,129 +748,87 @@ protected function get_default_primary_column_name() {
-					__( 'Edit' )
-				);
-			}
+		if ( ! $this->is_trash && current_user_can( 'delete_post', $post->ID ) && ( EMPTY_TRASH_DAYS && MEDIA_TRASH ) ) {
+			$actions['trash'] = sprintf(
+				'<a href="%s" class="submitdelete aria-button-if-js" aria-label="%s">%s</a>',
+				wp_nonce_url( "post.php?action=trash&amp;post=$post->ID", 'trash-post_' . $post->ID ),
+				/* translators: %s: Attachment title. */
+				esc_attr( sprintf( __( 'Move &#8220;%s&#8221; to the Trash' ), $att_title ) ),
+				_x( 'Trash', 'verb' )
+			);
+		}
 
-			if ( current_user_can( 'delete_post', $post->ID ) ) {
-				if ( $this->is_trash ) {
-					$actions['untrash'] = sprintf(
-						'<a href="%s" class="submitdelete aria-button-if-js" aria-label="%s">%s</a>',
-						wp_nonce_url( "post.php?action=untrash&amp;post=$post->ID", 'untrash-post_' . $post->ID ),
-						/* translators: %s: Attachment title. */
-						esc_attr( sprintf( __( 'Restore &#8220;%s&#8221; from the Trash' ), $att_title ) ),
-						__( 'Restore' )
-					);
-				} elseif ( EMPTY_TRASH_DAYS && MEDIA_TRASH ) {
-					$actions['trash'] = sprintf(
-						'<a href="%s" class="submitdelete aria-button-if-js" aria-label="%s">%s</a>',
-						wp_nonce_url( "post.php?action=trash&amp;post=$post->ID", 'trash-post_' . $post->ID ),
-						/* translators: %s: Attachment title. */
-						esc_attr( sprintf( __( 'Move &#8220;%s&#8221; to the Trash' ), $att_title ) ),
-						_x( 'Trash', 'verb' )
-					);
-				}
+		if ( $this->is_trash && current_user_can( 'delete_post', $post->ID ) || ! EMPTY_TRASH_DAYS || ! MEDIA_TRASH ) {
+			$actions['delete'] = sprintf(
+				'<a href="%s" class="submitdelete aria-button-if-js"%s aria-label="%s">%s</a>',
+				wp_nonce_url( "post.php?action=delete&amp;post=$post->ID", 'delete-post_' . $post->ID ),
+				! MEDIA_TRASH ? " onclick='return showNotice.warn();'" : '',
+				/* translators: %s: Attachment title. */
+				esc_attr( sprintf( __( 'Delete &#8220;%s&#8221; permanently' ), $att_title ) ),
+				__( 'Delete Permanently' )
+			);
+		}
 
-				if ( $this->is_trash || ! EMPTY_TRASH_DAYS || ! MEDIA_TRASH ) {
-					$delete_ays        = ( ! $this->is_trash && ! MEDIA_TRASH ) ? " onclick='return showNotice.warn();'" : '';
-					$actions['delete'] = sprintf(
-						'<a href="%s" class="submitdelete aria-button-if-js"%s aria-label="%s">%s</a>',
-						wp_nonce_url( "post.php?action=delete&amp;post=$post->ID", 'delete-post_' . $post->ID ),
-						$delete_ays,
-						/* translators: %s: Attachment title. */
-						esc_attr( sprintf( __( 'Delete &#8220;%s&#8221; permanently' ), $att_title ) ),
-						__( 'Delete Permanently' )
-					);
-				}
-			}
+		if ( ! $this->is_trash && get_permalink( $post->ID ) ) {
+			$actions['view'] = sprintf(
+				'<a href="%s" aria-label="%s" rel="bookmark">%s</a>',
+				get_permalink( $post->ID ),
+				/* translators: %s: Attachment title. */
+				esc_attr( sprintf( __( 'View &#8220;%s&#8221;' ), $att_title ) ),
+				__( 'View' )
+			);
+		}
```

Comment:
We can reduce the number of calls to `get_permalink()` here too:
```suggestion
		if ( ! $this->is_trash ) {
			$permalink = get_permalink( $post->ID );
			if ( $permalink ) {
				$actions['view'] = sprintf(
					'<a href="%s" aria-label="%s" rel="bookmark">%s</a>',
					$permalink,
					/* translators: %s: Attachment title. */
					esc_attr( sprintf( __( 'View &#8220;%s&#8221;' ), $att_title ) ),
					__( 'View' )
				);
			}
		}
```

Reply:
Oh also, shouldn't the permalink be passed through `esc_url()`?

Reply:
I think the permalink could be escaped through `esc_url()`. Do you think `get_edit_post_link()` and `wp_nonce_url()` should be escaped with it  too, the pattern exists in core/trunk. Any directions what to do here Sergey?